test/unit/SimulatePoolReserves.t.sol

```diff
@@ -7,111 +7,248 @@ import { GPv2Order } from "cowprotocol/contracts/libraries/GPv2Order.sol";
+
+            // Assertions
+            assertApproxEqRel(token0Bal, token0Reserve, 1e8); // 1e18 == 100%
+            assertApproxEqRel(token1Bal, token1Reserve, 1e8);
+
+            // One would expect the pool to be balanced after the trade adjustment.
+            // Verify approximately balanced pool reserves.
+            assertApproxEqRel(
+                token0Bal * price0,
+                token1Bal * price1,
+                1e8, // 1e18 == 100%
+                "Relative price calculation incorrect"
+            );
         }
     }
 
-    function relativeDiffBips(
-        uint256 token0Bal,
-        uint256 token1Bal,
-        uint256 price0,
-        uint256 price1
-    )
-        internal
-        pure
-        returns (uint256 bips)
-    {
-        uint256 value0 = token0Bal * price0;
-        uint256 value1 = token1Bal * price1;
-
-        // Calculate relative difference as BIPS (1 BIP = 0.01%)
-        if (value0 > value1) {
-            bips = ((value0 - value1) * 10_000) / value0;
-        } else {
-            bips = ((value1 - value0) * 10_000) / value1;
+    function test_SimulatePoolReserves_Imbalanced_50_50Pool_IncreasingToken1Amount() external {
+        uint256 token0Reserve = 100e18;
+        uint256 token1Reserve = 300_000e18;
+        uint256 price0 = 3000e8;
+        uint256 price1 = 1e8;
+
+        uint256[] memory token1Reserves = new uint256[](7);
+        token1Reserves[0] = 306_000e18;
+        token1Reserves[1] = 315_000e18;
+        token1Reserves[2] = 350_000e18;
+        token1Reserves[3] = 400_000e18;
+        token1Reserves[4] = 750_000e18;
+        token1Reserves[5] = 900_000e18;
+        token1Reserves[6] = 30_000_000e18;
+
+        for (uint256 i; i < token1Reserves.length; ++i) {
+            // Calculate token1AmountOut given token0Reserves: trading against pool
+            uint256 token0AmountOut = calcOutGivenInSignedWadMath(
+                token1Reserve,
+                defaults.WEIGHT_50(),
+                token0Reserve,
+                defaults.WEIGHT_50(),
+                token1Reserves[i] - token1Reserve
+            );
+
+            // Mock token balances
+            setTokenBalances(token0Reserve - token0AmountOut, token1Reserves[i]);
+
+            // Simulate pool reserves
+            (uint256 token0Bal, uint256 token1Bal) = oracle.exposed_simulatePoolReserves(price0, price1);
+
+            // Assertions
+            assertApproxEqRel(token0Bal, token0Reserve, 1e8); // 1e18 == 100%
+            assertApproxEqRel(token1Bal, token1Reserve, 1e8);
+
+            // One would expect the pool to be balanced after the trade adjustment.
+            // Verify approximately balanced pool reserves.
+            assertApproxEqRel(
+                token0Bal * price0,
+                token1Bal * price1,
+                1e8, // 1e18 == 100%
+                "Relative price calculation incorrect"
+            );
         }
     }
+
+    // @todo: @bh2smith - is this test still helpful?
```

Comment:
May not be to the extent that's it's done here. As long as we have a test demonstrating how imbalanced pools sway away from expectations the further they get. Ideally we have a curve fitted so we don't need all the data points. 

Oh. I see the test above seems to cover this.